lib/taskana-core-test/src/test/java/acceptance/task/requestreview/RequestReviewWithBeforeSpiAccTest.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    @Override
+    public Task beforeRequestReview(Task task) throws Exception {
+      task.setCustomField(TaskCustomField.CUSTOM_3, NEW_CUSTOM_3_VALUE);
+      task = taskanaEngine.getTaskService().updateTask(task);
+      return task;
+    }
+  }
+
+  static class ClassificationUpdater implements BeforeRequestReviewProvider {
+
+    public static final String SPI_CLASSIFICATION_NAME = "Neuer Classification Name";
+
+    private TaskanaEngine taskanaEngine;
+
+    @Override
+    public void initialize(TaskanaEngine taskanaEngine) {
+      this.taskanaEngine = taskanaEngine;
+    }
+
+    @Override
+    public Task beforeRequestReview(Task task) throws Exception {
+      Classification newClassification =
+          defaultTestClassification().buildAndStore(taskanaEngine.getClassificationService());
+
+      task.setClassificationKey(newClassification.getKey());
+      task = taskanaEngine.getTaskService().updateTask(task);
+
+      newClassification.setName(SPI_CLASSIFICATION_NAME);
+      taskanaEngine.getClassificationService().updateClassification(newClassification);
+
+      return task;
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = BeforeRequestReviewProvider.class,
+      serviceProviders = TaskModifier.class)
+  class SpiModifiesTask {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskClaimedByUser("user-1-1").buildAndStore(taskService);
+
+      Task result = taskService.requestReview(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3))
+          .isEqualTo(TaskModifier.NEW_CUSTOM_3_VALUE);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnPersistentModifiedTask_When_SpiModifiesAndTransfersTask() throws Exception {
+      Task task = createTaskClaimedByUser("user-1-1").buildAndStore(taskService);
+
+      taskService.requestReview(task.getId());
+      Task result = taskService.getTask(task.getId());
+
+      assertThat(result.getCustomField(TaskCustomField.CUSTOM_3))
+          .isEqualTo(TaskModifier.NEW_CUSTOM_3_VALUE);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  @WithServiceProvider(
+      serviceProviderInterface = BeforeRequestReviewProvider.class,
+      serviceProviders = SetTaskOwner.class)
+  class SpiSetsValuesWhichGetOverridenByTaskana {
+
+    @TaskanaInject TaskService taskService;
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void name() throws Exception {
+      Task task = createTaskClaimedByUser("user-1-1").buildAndStore(taskService);
+
+      taskService.forceRequestReview(task.getId());
+      Task result = taskService.getTask(task.getId());
+
+      assertThat(result.getOwner()).isNull();
+    }
+
+    class SetTaskOwner implements BeforeRequestReviewProvider {
```

Comment:
Why is this SPI not being placed together with other SPIs?

Reply:
Because it is needed only in the enclosing test class. 
You prefer that the SPI's are located together?